core/app/models/spree/fulfilment_changer.rb

```diff
@@ -114,6 +114,14 @@ def handle_stock_counts?
       current_shipment.order.completed? && current_stock_location != desired_stock_location
     end
 
+    def default_on_hand_quantity
+      if current_stock_location != desired_stock_location
+        0
+      else
+        current_shipment.inventory_units.where(variant: variant).on_hand.limit(quantity).count
```

Comment:
Why the `limit(quantity)` is needed here? Shouldn't `current_shipment.inventory_units.where(variant: variant).on_hand` already return only the inventory units present in the shipment for this variant, which should not be more than `quantity`?

Reply:
@kennyadsl I think your note makes sense, that code is not necessary. I'm changing the code soon.